contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpStoragePluginConfig.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.http;
+
+import org.apache.drill.common.map.CaseInsensitiveMap;
+import org.apache.drill.common.logical.StoragePluginConfigBase;
+import org.apache.drill.shaded.guava.com.google.common.base.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+
+@JsonTypeName(HttpStoragePluginConfig.NAME)
+public class HttpStoragePluginConfig extends StoragePluginConfigBase {
+  private static final Logger logger = LoggerFactory.getLogger(HttpStoragePluginConfig.class);
+
+  public static final String NAME = "http";
+
+  public final Map<String, HttpAPIConfig> connections;
+
+  public final boolean cacheResults;
```

Comment:
The OkHTTP library has a caching mechanism [1] which caches HTTP and HTTPS responses to the filesystem so they may be reused, saving time and bandwidth.  These results are stored to the Drill temp directory.  

I saw this feature in OkHTTP and it seemed like a good idea at the time to allow the user to enable/disable it if it makes sense.  The use case I had in mind was querying some relatively unchanging reference data via API. I was thinking that it would make sense to minimize the number of HTTP calls and cache the data locally for a period of time. Again, the library handles the caching logic, so in theory we don't have to worry about that.

[1] https://square.github.io/okhttp/4.x/okhttp/okhttp3/-cache/